File: it.unibo.bls19Local/src/main/java/it/unibo/blsFramework/interfaces/IBlsFramework.java
package it.unibo.blsFramework.interfaces;

import it.unibo.bls.interfaces.ILed;
import it.unibo.bls.interfaces.IObservable;
import it.unibo.bls.interfaces.IObserver;

public interface IBlsFramework {
    public void addConcreteButton(IObservable button);
    public void setConcreteLed( ILed led );
    public void addConcreteLed( ILed led );
    public void setApplLogic(  IAppLogic appLogic );
}

File: it.unibo.bls19Local/src/main/kotlin/it/unibo/bls/kotlin/applLogic/IBlsFramework.kt
package it.unibo.bls.kotlin.applLogic

import it.unibo.bls.interfaces.IObservable
import it.unibo.bls.interfaces.IObserver
import it.unibo.blsFramework.interfaces.IAppLogic

interface IBlsFramework {
    fun setConcreteLed(led: IObserver)
    fun setConcreteButton(button: IObservable)
    fun setApplLogic(  arg : IAppLogic)
}
